<?xml version="1.0"?>
<launch>
  <!-- FER-LAMOR -->

  <arg name="h1_x" default="0.0"/>
  <arg name="h1_y" default="0.0"/>
  <arg name="h1_yaw" default="0.0"/>
  <arg name="h2_x" default="0.0"/>
  <arg name="h2_y" default="0.0"/>
  <arg name="h2_yaw" default="0.0"/>

  <group ns="/husky1"> <!-- ne smijes ovo maknut jer ti nece bit frameovi od drugog huskyja nego ce bit kolizija -->
  <!-- <group> -->
    <param name="tf_prefix" value="husky1_tf" />
    <include file="$(find husky_dynamic_navigation)/launch/spawn_single_robot.launch">
      <!-- <env name="ROBOT_NAMESPACE" value="/husky1"/>
      <env name="HUSKY_UST10_ENABLED" value="1"/> 
      <env name="HUSKY_UST10_TOPIC" value="/husky1/front/scan"/> ovo uopce ne setupa env-ove ?? -->
      <arg name="joystick" value="false" />
      <arg name="robot_namespace" value="/husky1"/>
      <arg name="husky_num" value="husky1" />
      <arg name="x" value="$(arg h1_x)"/> 
      <arg name="y" value="$(arg h1_y)"/>
      <arg name="z" value="0.0"/>
      <arg name="yaw" value="$(arg h1_yaw)"/>
      <arg name="model" value="husky1"/>
    </include>
  </group>
  
  <group ns="/husky2">
  <!-- <group> -->
    <param name="tf_prefix" value="husky2_tf" />
    <!-- <include file="$(find husky_gazebo)/launch/spawn_husky.launch"> -->
    <include file="$(find husky_dynamic_navigation)/launch/spawn_single_robot.launch">
      <!-- <env name="ROBOT_NAMESPACE" value="/husky2"/>
      <env name="HUSKY_UST10_TOPIC" value="/husky2/front/scan"/>  -->
      <arg name="joystick" value="false" />
      <arg name="robot_namespace" value="/husky2"/>
      <arg name="husky_num" value="husky2" />
      <arg name="x" value="$(arg h2_x)"/>
      <arg name="y" value="$(arg h2_y)"/>
      <arg name="z" value="0.0"/>
      <arg name="yaw" value="$(arg h2_yaw)"/>
      <arg name="model" value="husky2"/>
    </include>
  </group>
</launch>